lib/iris/_merge.py

```diff
@@ -1586,18 +1614,34 @@ def _build_coordinates(self):
                               self._vector_aux_coords_dims):
             aux_coords_and_dims.append(_CoordAndDims(item.coord, dims))
 
-    def _build_signature(self, cube):
-        """Generate the signature that defines this cube."""
+    def _build_signature(self, cube, fill_value=None):
+        """
+        Generate the signature that defines this cube.
+
+        Args:
+
+        * cube:
+            The source cube to create the cube signature from.
+
+        Kwargs:
+
+        * fill_value:
+            Promoted fill-value to use instead of the source cube
+            fill-value.
+
+        Returns:
+            The cube signature.
+
+        """
+        if fill_value is None:
+            fill_value = cube.fill_value
         return _CubeSignature(cube.metadata, cube.shape, cube.dtype,
-                              cube.fill_value, cube._cell_measures_and_dims)
+                              fill_value, cube._cell_measures_and_dims)
```

Comment:
@dkillick This change support the need to rebuilt the proto-cube cube signature in the event of a `fill_value` promotion.